http-ballerina/http_request.bal

```diff
@@ -389,7 +390,7 @@ public class Request {
                             string value = entries[entryIndex].substring(index + 1, size);
                             value = value.trim();
                             if (value != "") {
-                                parameters[name] = value;
+                                parameters[check decode(name)] = check decode(value);
```

Comment:
* Do we have to do url-decode for each entry? Can't we just decode the whole payload once and then split it with `&` ?